net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/checkstyle-metadata.properties

```diff
@@ -68,4 +68,7 @@ UniqueProperties.fileExtensions = file type extension of the files to check.
 
 SuppressWarningsHolder.name = SuppressWarningsHolder
 SuppressWarningsHolder.desc = Maintains a set of check suppressions from {@link SuppressWarnings} annotations.
-SuppressWarningsHolder.aliasList = the list of comma-separated alias assignments
\ No newline at end of file
+SuppressWarningsHolder.aliasList = the list of comma-separated alias assignments
+
+NoCodeInFile.name = No Code In File
+NoCodeInFile.desc = Checks whether file contains code.<br/><br/>Files which are considered to have no code: <ul><li>Files with no text</li><li>Files with only single line or multi-line comment(s)</li></ul>
```

Comment:
\n

Reply:
Using \n instead of HTML break leads to no line break being rendered, see here:
![noCodeInFile](https://user-images.githubusercontent.com/406876/83379490-ef872c00-a3db-11ea-87d0-fa37894012da.png)
Also the file already contains HTML line breaks everywhere, e.g. just some lines above the mentioned location at https://github.com/checkstyle/eclipse-cs/blob/5e578a991d07d0d5b02faf5b84c51610df2d83aa/net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/checkstyle-metadata.properties#L66

![ArrayTrailingComma](https://user-images.githubusercontent.com/406876/83379521-0463bf80-a3dc-11ea-8de5-9e3881a940f2.png)

Results of the checks (on the actual tests contained in the checkstyle sources):
![2020-06-01_07h42_36](https://user-images.githubusercontent.com/406876/83379531-09c10a00-a3dc-11ea-9ef0-35836e274359.png)
![NoCodeInFile2](https://user-images.githubusercontent.com/406876/83379541-10e81800-a3dc-11ea-8337-a4a1181d78c8.png)

To me it seems to be working fine

Reply:
@Bananeweizen , in screenshot I do not see problem. Please point by marker what become bad.
All I care is extra new line symbol to avoid this problem:
![Screen Shot 2020-06-09 at 1 14 12 PM](https://user-images.githubusercontent.com/812984/84195422-6616d000-aa53-11ea-9b83-ad06fa51d7f8.png)

we have Check on this - https://checkstyle.org/config_misc.html#NewlineAtEndOfFile (one day might be enforced over code base)

Reply:
Ok, sorry about it, this is very good best practice especially for non java files. I was very curious about issue you raised, as  we now in progress to generate such meta files, so I am very interested in any nuances in such files.